source/classes.tex

```diff
@@ -3305,16 +3305,17 @@
 \pnum
 \indextext{\idxcode{union}!global anonymous}%
 \indextext{scope!anonymous \tcode{union} at namespace}%
-Anonymous unions declared in a named namespace or in the global
-namespace shall be declared \tcode{static}. Anonymous unions declared at
-block scope shall be declared with any storage class allowed for a
-block-scope variable, or with no storage class. A storage class is not
-allowed in a declaration of an anonymous union in a class scope.
+Anonymous unions declared in a named namespace or in the global namespace
+shall be declared \tcode{static}.
+A block-scope anonymous union declaration shall not
+have a \grammarterm{storage-class-specifier} that is not
+permitted in the declaration of a block-scope variable.
+A \grammarterm{storage-class-specifier} shall not
+appear in a class-scope anonymous union declaration.
```

Comment:
I would prefer "have" over "appear in" here -- "A class-scope anonymous union declaration shall not have a *storage-class-specifier*." That would both parallel the previous two sentences better, and also avoids the suggestion that we might be referring to *storage-class-specifier*s nested in the union members somehow.